src/Tools/dotnet-monitor/Extensibility/ExtensionMode.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    [JsonConverter(typeof(JsonStringEnumConverter))]
```

Comment:
Could you see if moving this attribute to the `ExtensionManifest.Modes` property works? This way, there isn't an indication or expectation from the extension implementation that this is involved in JSON serialization (even though it is for this one specific usage).